db/app.js

```diff
@@ -1,14 +1,25 @@
 const express = require("express");
-const { getTopics } = require("./controller/controller.js");
+const { getTopics, getArticleById } = require("./controller/controller.js");
 const app = express();
 
 app.get("/api/topics", getTopics);
 
+app.get('/api/articles/:article_id', getArticleById)
+
 app.all('/api/*', (req, res, next) => {
     res.status(404).send({ msg: 'path not found'})
   })
 
-
+app.use((err, req, res, next) => {
+    if(err.status){
+     res.status(err.status).send({ msg: err.msg})
+    }
+  next(err)
+     })
+  
+  app.use((err, req, res, next) => {
+   res.status(400).send({ msg: 'Bad request'})
+  })
```

Comment:
the last error handler in the chain should idealy be the 500 internal server error handler.
If this is what you're using to try and catch the error of not a valid ID then look at what PSQL throws back when you give it something other than a number and this will be useful to you for handling errors.